packages/api/src/providers/fitbit.ts

```diff
@@ -373,7 +388,86 @@ export class Fitbit extends Provider implements OAuth2 {
     );
   }
 
-  // Creates a subscription for all or one specific collectionType
+  // Finds existing users connected to this Fitbit account, revokes their tokens and sends WH updates about the disconnections
+  async revokeTokenFromExistingUsers(
+    userId: string,
+    fitbitUserId: string,
+    activeSubscriptions: string[],
+    accessToken: string
+  ) {
+    try {
+      if (activeSubscriptions.length > 0) {
+        const connectedUsers = await getConnectedUsersByTokenOrFail(
+          ProviderSource.fitbit,
+          fitbitUserId
+        );
+        await Promise.all(
+          Object.values(connectedUsers).map(async user => {
+            if (user.dataValues.id !== userId) {
+              await this.oauth.revokeLocal(user);
+              user = await getConnectedUserOrFail({
+                id: user.id,
+                cxId: user.cxId,
+              });
+              sendProviderDisconnected(user, [ProviderSource.fitbit]);
+            }
+          })
+        );
+        await this.deleteSubscriptions(activeSubscriptions, fitbitUserId, accessToken);
+      }
+    } catch (err) {
+      console.log("Failed to revoke the token of a Fitbit user.", err);
+      capture.error("Failed to revoke the token of a Fitbit user.", {
+        extra: { context: "fitbit.revokeTokenFromExistingUsers" },
+      });
+    }
+  }
+
+  // Deletes all WH subscriptions for a user
+  async deleteSubscriptions(
+    activeSubscriptions: string[],
+    fitbitUserId: string,
+    accessToken: string
+  ): Promise<void> {
+    try {
+      await Promise.all(
+        activeSubscriptions.map(async subscriptionType => {
```

Comment:
I suggest we name these variables/parameters consistently. From what I could tell from this PR, those all store the same information (subscription collection types), right?
- `activeSubscriptions`
- `subscriptionType`
- `subscriptions`